### PR TITLE
chore: upgrade to bitcoinjs-lib v6

### PR DESCRIPTION
**Pull request by motorina0**
## Summary
 - update dependency for `bitcoinjs-lib` to `v6`
 - fix one broken test (see comment  for commit https://github.com/bitcoinjs/bolt11/commit/b34dd0a23d0a051f824eb4719477c5cb09ba4332)
 - parser now allows `fallback_address.code` to be 1
 - add unit tests for witness v1 `fallback_adress`

Closes: https://github.com/bitcoinjs/bolt11/issues/52

### CODE REVIEW

File: test/fixtures.json [hunk 1]
```diff
@@ -518,7 +518,7 @@
             {
               "tagName": "fallback_address",
               "data": {
-                "address": "tb1dqqqsyqcyq5rqwzqfpg9scrgwpuqsyqcyy5mynw"
+                "address": "tb1zx93njcnp8qcnqde4vgckydpkv33rsdnpxccrwetp8qmn2vnrv4jnjdfs8q6rsdpksh0hkj"
```
Comment (junderw) [inline test/fixtures.json:521]: Why was this changed?

**Review by junderw — APPROVED** [review]
LGTM

File: test/fixtures.json (hunk 1 above)
Comment (motorina0) [reply test/fixtures.json:521]: The bitcoinjs-lib address validation is more strict in v6. It requires:
- version === 0 for bech32
- version !== 0 for bech32m
This fix uses an address that has version === 2 with bech32m encoding

**Comment by junderw** [comment]
Also, don't forget package version bump in lock file too.

**Comment by motorina0** [comment]
> Also, don't forget package version bump in lock file too.

Done!

File: test/fixtures.json (hunk 1 above)
Comment (motorina0) [reply test/fixtures.json:521]: With the old `address` value plus `v6` an exception was thrown earlier (`'Fallback address type is unknown'`), and the code did get the chance to check and fail with `Fallback address witness version is unknown`

**Comment by junderw** [comment]
Published to v1.3.3

secp256k1 dependency was having trouble building on v14 so I updated it.

**Comment by motorina0** [comment]
> Published to v1.3.3
> 
> secp256k1 dependency was having trouble building on v14 so I updated it.

hmmm, tests fail for me with:

>   const sigObj = secp256k1.sign(payReqHash, privateKey)
>                            ^
> 
> TypeError: secp256k1.sign is not a function
>     at Object.sign (/Users/moto/Documents/GitHub/bitcoincoretech/bolt11/payreq.js:533:28)
>     at Test.<anonymous> (/Users/moto/Documents/GitHub/bitcoincoretech/bolt11/test/index.js:89:33)

**Comment by junderw** [comment]
oops

my bad.

published v1.3.4
